armi/utils/gridEditor.py

```diff
@@ -0,0 +1,1783 @@
+# USE AND DISTRIBUTION OF THIS CODE IS GOVERNED BY EXPORT CONTROL LAWS AND THE LICENSE IN LICENSE.txt.
+# @LICENSE:
+"""
+GUI elements for manipulating core layout.
+
+This provides a handful of classes which provide wxPython Controls for manipulating
+grids and grid Blueprints.
+
+Known Issues
+============
+
+ * There is no action stack or undo functionality. Save frequently if you want to
+ recover previous states
+
+ * Cartesian grids are supported, but not rendered as nicely as their Hex counterparts.
+ The "through center assembly" case is not rendered properly with the half-assemblies
+ that lie along the edges.
+
+ * The controls are optimized for manipulating a Core layout, displaying an "Assembly
+ palette" that contains the Assembly designs found in the top-level blueprints. A little
+ extra work and this could also be made to manipulate block grids or other things.
+
+ * Assembly colors are derived from the set of flags applied to them, but the mapping of
+ colors to flags is not particularly rich, and there isn't anything to disambiguate
+ between asemblies of different design, but the same flags.
```

Comment:
ok. nice to have it listed here. I suspect someone will enhance this to solve this issue relatively quickly but it's fine to just list it here and move on.